lib/Ohmbrewer_Thermostat.cpp

```diff
@@ -1,93 +1,99 @@
-    if ( n > 2){
-        _heatingElm = new HeatingElement(id+2, thermPins[1], thermPins[2]);
-        _tempSensor = new TemperatureSensor(id+1, thermPins[0]);
-    }else if (n == 2 ){                     // no switch, only power% (SSR in parallel or always on other leg)
-        _heatingElm = new HeatingElement(id+2, thermPins[1], -1);
-        _tempSensor = new TemperatureSensor(id+1, thermPins[0]);
-    }else{
-        //publish error
-
+Ohmbrewer::Thermostat::Thermostat(int id, std::list<int>* thermPins) : Ohmbrewer::Equipment(id) {
+    //PID set up
+    thermPID = (new PID(&input, &output, &setPoint, consKp, consKi, consKd, PID::DIRECT));
+    //initialize the variables we're linked to
+    windowStartTime = millis();
+    setPoint = _targetTemp->c();
+    input = _tempSensor->getTemp()->c();
+    //tell the PID to range between 0 and the full window size
+    thermPID->SetOutputLimits(0, windowSize);
+    //turn the PID on
+    thermPID->SetMode(PID::AUTOMATIC);
+
+    //initilize equipment components
+    int size = thermPins->size();
+    int tempB;
+    if ( (size == 2) || (size == 3) ){
+
+        tempB = thermPins->front();
+        thermPins->pop_front();
+        _heatingElm = new HeatingElement(id+3, thermPins);
+        _tempSensor = new TemperatureSensor(id+1, tempB);
+        _safetySensor = new TemperatureSensor(id+2, tempB);  //need UIDS
+    }else{//publish error
         Ohmbrewer::Publisher::publish_map_t pMap;
         Ohmbrewer::Publisher* pub = new Ohmbrewer::Publisher(new String("error_log"), &pMap);
-
-        pMap[String("array_check_thermostat")] = String("improperly formed array - Thermostat(int id, int[])");
+        pMap[String("list_check_thermostat")] = String("improperly formed input - Thermostat(int id, int<list>)");
         pub->publish();
     }
-
-    _targetTemp = new Temperature(0);
+    _targetTemp = new Temperature(-69);
     registerUpdateFunction();
 }
 
 /**
  * Constructor
  * @param id The Sprout ID to use for this piece of Thermostat
- * @param tubePins[ temp busPin ; heating powerPin ; heating controlPin ]
+ * @param thermPins list with formatting of: [ temp busPin ;  heating controlPin ; heating powerPin ]
  * @param targetTemp The new target temperature in Celsius
  */
-Ohmbrewer::Thermostat::Thermostat(int id, int (&thermPins)[3], const double targetTemp) : Ohmbrewer::Equipment(id) {
-    int n = sizeof(thermPins) / sizeof(int);
-    if ( n > 2){
-        _heatingElm = new HeatingElement(id+2, thermPins[1], thermPins[2]);
-        _tempSensor = new TemperatureSensor(id+1, thermPins[0]);
-    }else if (n == 2 ){                     // no switch, only power% (SSR in parallel or always on other leg)
-        _heatingElm = new HeatingElement(id+2, thermPins[1], -1);
-        _tempSensor = new TemperatureSensor(id+1, thermPins[0]);
-    }else{
-        //publish error
+Ohmbrewer::Thermostat::Thermostat(int id, std::list<int>* thermPins, const double targetTemp) : Ohmbrewer::Equipment(id) {
```

Comment:
All the stuff I mentioned for the previous constructor applies there too, naturally.